doc/ocaml-ecosystem.rst

```diff
@@ -0,0 +1,89 @@
+The OCaml Ecosystem
+===================
+
+The OCaml ecosystem is not monolithic: the compiler and tools are not
+maintained by the same entities. As such, it can be difficult to understand the
+history and roles of the various pieces of this ecosystem. The goal of this
+page is to give a quick overview of the situation and of the role that Dune
+plays in it.
+
+The Compiler Distribution: Compiling and Linking
+------------------------------------------------
+
+The compiler distribution contains "core" tools including the compilers
+(``ocamlc`` and ``ocamlopt``). They turn source files (with extensions ``.ml``
+and ``.mli``) into executables and libraries. Dependencies between compiled
+objects only exist at the module level, so this is a low-level tool.
+
+Findlib: Metadata for libraries
+-------------------------------
+
+Findlib is a tool that defines the concept of library, so that libraries can
+depend on other libraries on top of the notion of module. Definitions of
+libraries, and other pieces of metadata, are stored in ``META`` files.
+
+Findlib ships an executable named ``ocamlfind`` that can be used as a wrapper
+on top of the compilers to perform tasks such as producing an executable from
+compiled object files and external libraries.
+
+Opam: a Collection of Software Projects
+---------------------------------------
+
+Opam is a package manager. It is used to determine which packages are
+necessary, and how to fetch and build them. Packages can contain libraries,
+executables, and other kinds of files.
+
+The notion of version is specific to opam. If your project uses a function
+named ``Png.read_file`` but this function has been added only in version
+``1.2.0`` of that package, opam needs to know about it.
+
+Opam manages collections of installed packages, called switches. Using your
+project's dependencies (names and version constraints), it is able to create a
+switch that you'll be using to develop your project.
+
+Public definition of packages are available in a database called
+``opam-repository`` which is maintained as a public Git repository. Publishing a
+package on opam (to make sure that external users can use your project)
+consists in adding its definition to ``opam-repository``.
```

Comment:
```suggestion
consists of adding its definition to ``opam-repository``.
```

Reply:
I'm not a native speaker, but I don't think that this is correct. publishing a package is not made up of adding its definition to ``opam-repository``, but publishing a package is essentially adding its definition to ``opam-repository``. @christinerose is this the right distinction?